public/js/viewports.js

```diff
@@ -426,34 +224,336 @@ var Viewport = function(peerName, dashboard) {
+
+        switch (viewports) {
+        case 0:
+            layout = { rows: 0, grid: [] };
+            break;
+        case 1:
+            layout = { rows: 1, grid: [1] };
+            break;
+        case 2:
+            layout = { rows: 1, grid: [2] };
+            break;
+        case 3:
+            layout = { rows: 1, grid: [3] };
+            break;
+        case 4:
+            layout = { rows: 2, grid: [2, 2] };
+            break;
+        case 5:
+            layout = { rows: 2, grid: [2, 3] };
+            break;
+        case 6:
+            layout = { rows: 2, grid: [3, 3] };
+            break;
+        case 7:
+            layout = { rows: 2, grid: [3, 4] };
+            break;
+        case 8:
+            layout = { rows: 2, grid: [4, 4] };
+            break;
+        case 9:
+            layout = { rows: 3, grid: [3, 3, 3] };
+            break;
+        case 10:
+            layout = { rows: 3, grid: [3, 4, 3] };
+            break;
+        case 11:
+            layout = { rows: 3, grid: [4, 3, 4] };
+            break;
+        case 12:
+            layout = { rows: 3, grid: [4, 4, 4] };
+            break;
+        case 13:
+            layout = { rows: 3, grid: [4, 5, 4] };
+            break;
+        case 14:
+            layout = { rows: 3, grid: [5, 4, 5] };
+            break;
+        case 15:
+            layout = { rows: 3, grid: [5, 5, 5] };
+            break;
+        default:
+            break;
+        }
+
+        return layout;
+    };
+
+    /**
+     * Returns the grid configuration for hangouts mode.
+     *
+     * @returns {Object} Grid configuration object.
+     * @public
+     */
+    this.gridForHangoutsMode = function() {
+        if (!this.viewportArray.length) {
+            return { rows: 2, grid: [0, 0] };
+        }
+
+        return { rows: 2, grid: [1, this.viewportArray.length - 1] };
+    };
+
+    /**
+     * Shows the viewports in dash mode.
+     *
+     * @returns {undefined} undefined
+     * @public
+     */
+    this.showDashMode = function() {
+        this.hangoutsMode = false;
+        if (this.viewportArray.length > 1) {
+            this.placeViewports();
+        }
+    };
+
+    /**
+     * Shows the viewports in hangout mode.
+     *
+     * @returns {undefined} undefined
+     * @public
+     */
+    this.showHangoutsMode = function() {
+        this.hangoutsMode = true;
+        if (this.viewportArray.length > 1) {
+            this.placeViewports();
+        }
+    };
+
+    /**
+     * Generates a dashboard grid for a new user.
+     *
+     * @param {String} peerName - The peer name of the
+     * user for which we should create a grid.
+     * @returns {Object} A new Viewport object.
+     * @public
+     */
+    this.createGridForNewUser = function(peerName) {
+        var newViewport = new Viewport(peerName, this);
+
+        // If peerName is undefined, this means the viewport is for the user themselves.
+        if (peerName === undefined) {
```

Comment:
I think `if (peerName)` can be OK since the intention is just to make sure that this condition is hit only if `peerName` is a populated string. I would have to double-check (test the code out dynamically).